method_comparison/peft_bench/README.md

```diff
@@ -0,0 +1,216 @@
+2. **Experiment-specific overrides** (`benchmark_params.json` in each experiment) - Optional overrides for specific experiments  
+3. **Adapter configuration** (`adapter_config.json` in each experiment) - PEFT method parameters
+
+This structure ensures consistent comparison while allowing flexibility where needed.
+
+### Default Configuration (`default_benchmark_params.json`)
+
+Contains shared benchmark settings that apply to all experiments. Here are the key configuration fields:
+
+- `model_id`: The Hugging Face model ID to use as the base model (e.g., "facebook/opt-350m")
+- `peft_method`: The PEFT method to apply (e.g., "lora", "prefix_tuning", etc.)
+- `dtype`: Model precision ("float16", "float32", or "bfloat16")
+- `seed`: Random seed for reproducibility
+- `max_new_tokens`: Maximum number of tokens to generate during inference
+- `num_inference_runs`: Number of inference runs per prompt for statistical reliability
+- `train_batch_size`: Batch size for training
+- `train_steps`: Number of training steps
+- `num_prompt_samples`: Number of prompt samples to use per category (short/medium/long)
+
+Each experiment can override these settings by providing its own `benchmark_params.json` file.
+
+### Experiment Structure
+
+Each experiment directory should contain:
+
+1. `adapter_config.json`: PEFT adapter configuration. For details on available parameters and their meanings, refer to the [PEFT documentation](https://huggingface.co/docs/peft/main/en/developer_guides/adapters).
+
+2. (Optional) `benchmark_params.json`: Override specific benchmark parameters for this experiment.
+
+Example directory structure:
+```
+experiments/
+└── lora/
+    ├── lora_r8/                # LoRA rank 8 experiment
+    │   ├── adapter_config.json # PEFT adapter configuration
+    │   └── benchmark_params.json # Optional benchmark overrides
+    └── lora_r16/               # LoRA rank 16 experiment
+        └── adapter_config.json
+```
+
+### Experiment-Specific Overrides Example
+
+If an experiment needs different benchmark settings, create `benchmark_params.json`:
+```json
+{
+    "_comment": "Override settings for this specific experiment",
+    "max_new_tokens": 50,
+    "num_inference_runs": 15,
+    "num_prompt_samples": 2
+}
+```
+
+These parameters will override the defaults from `default_benchmark_params.json`.
+
+### Configuration Loading Process
+
+The benchmark runner follows this process:
+
+1. **Load defaults**: Read `default_benchmark_params.json` from the peft_bench directory
+2. **Check for overrides**: Look for `benchmark_params.json` in the experiment directory  
+3. **Merge configurations**: Override default values with experiment-specific values
+4. **Load adapter config**: Read the required `adapter_config.json` for PEFT parameters
+
+This ensures all experiments share the same baseline configuration unless explicitly overridden.
+
+### Adapter Configuration Example
+
```

Comment:
Instead of showing an example JSON, I'd rather show how to _create_ a new experiment, like is done here: https://github.com/huggingface/peft/tree/main/method_comparison/MetaMathQA#adapter_configjson